server/webapp/WEB-INF/rails.new/app/presenters/api_v1/scms/configuration_property_representer.rb

```diff
@@ -0,0 +1,57 @@
+##########################################################################
+# Copyright 2016 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  module Scms
+    class ConfigurationPropertyRepresenter < BaseRepresenter
+      alias_method :configuration_property, :represented
+
+      property :secure, skip_render: true, exec_context: :decorator
+      property :configKeyName, as: :key,
+               setter: lambda {|value, options|
+                 self.setConfigurationKey(ConfigurationKey.new(value))
+               }
+      property :value, skip_nil: true,
+               exec_context: :decorator
+      property :encrypted_value, skip_nil: true, exec_context: :decorator
+
+      def secure
+        configuration_property.isSecure
+      end
+
+      def secure=(value)
+        @secure = value
+      end
+
+      def value
+        configuration_property.getConfigurationValue.getValue if !configuration_property.isSecure
+      end
+
+      def value=(value)
+        configuration_property.setConfigurationValue(ConfigurationValue.new(value))
+        configuration_property.handleSecureValueConfiguration(@secure)
```

Comment:
You're dependent on the fact that `secure=` is called before `value=` which is not a guarantee that we can make, given that we don't know in what order the methods are invoked.

For things like these where the value of something is dependant on the value of another value see `EnvironmentVariableRepresenter#from_hash`